lib/cell.rb

```diff
@@ -23,10 +23,10 @@ def fired_upon?
   end
 
   def fire_upon
-    @hit = true
-    if @ship != nil
+    if @hit == false && @ship != nil
```

Comment:
That was a big problem before we totally don't want ta ship to lose health if hit on the same cell twice. This might be an unneeded change now that we fixed the valid turn method.